### DISCUSSION TITLE
Interactivity with charts + dashboards

### DISCUSSION THREAD

**Discussion by TuringLovesDeathMetal** [lightdash/lightdash#1138] [Ideas]
### ❓ What questions do we have for our users/community?
- When was the last time you used a dashboard filter? What did you use it for? 
- What's your favourite chart/dashboard interactivity feature of your current (or past) BI tool? 

### 💡 What’s the idea?

Things like click-to-filter, drill-down and dashboard filters. Some people have said they wouldn’t switch BI tools until this exist.

Dashboard level filters have come up a lot in feedback. Interactivity has come up with bigger users who are looking to switch.

E.g. I have a chart with top 10 products by number of sales and under that I have total spend per day. If I click on boots, it would filter all of the other charts to boots.

Drill-down hasn’t been requested as much as the dashboard filters. This is for any chart (both saved charts + saved charts in dashboards). If you click on an item in a chart, it shows you all of the values in that chart item (e.g. total number of users by status - click ‘active’, would show you all of the user IDs)

**Dashboard-level filters** is the highest priority. 

**Drill down into charts** is the next highest priority. 

**Click to filter** also feels like it fits in with the other two. 

### 📝 What’s the rough overview of work required to ship this?

*do we need user research? do we need to plan/partner with another company? do we need to find a new tool to do the thing we want to do?*

- Research into other BI tool interactivity and what the possibilities are.
    - Get examples (especially visual ones) of what this could look like in the app + how people would interact with this (what flows are there?)
    - Talk to the people who were looking for this feature (dashboard filters)
- Deciding what’s the MVP version of this big feature? (this should just be a written proposal with words + some pictures)
- Design
- Engineering

### ⏭ What are the next steps?

- Research!

### 😎 Any other notes/comments/important info

- It feels like these three features are all related - they feel like different ways of interacting with dashboard-level filters (or chart level filters).

**Comment by PriPatel** [comment]
**Questions for the community!**
❤️  Any insight welcome!!❤️


⏱️   When was the last time you used a dashboard filter? What did you use it for?  
🤔   What are some common filters that you use? eg. date? location?
🪰   How often do you apply these filters? Are they on-the-fly, or part of a routine?
🏆  What's your favourite chart/dashboard interactivity feature of your current (or past) BI tool?
